test/tools/Modules/HttpListener/HttpListener.psm1

```diff
@@ -0,0 +1,168 @@
+    .Description
+        Sends `exit` command to HTTP Listener causing it to exit.
+    .Parameter Port
+        Port to use, default is 8080
+    .Example
+        Stop-HTTPListener -Port 8080
+    #>
+    Param (
+        [Parameter()]
+        [Int] $Port = 8080
+    )
+
+    Invoke-WebRequest -Uri "http://localhost:$port/PowerShell?test=exit"
+}
+
+Function Start-HTTPListener {
+    <#
+    .Synopsis
+        Creates a new HTTP Listener to be used for PowerShell tests rather than rely on 3rd party websites
+    .Description
+        Creates a new HTTP Listener supporting several test cases intended to be used only by PowerShell tests.
+        This cmdlet requires running from an elevated administrator prompt to open a port.
+
+        Use Ctrl-C to stop the listener.  You'll need to send another web request to allow the listener to stop since
+        it will be blocked waiting for a request.
+    .Parameter Port
+        Port to listen, default is 8080
+    .Example
+        Start-HTTPListener -Port 8080
+        Invoke-WebRequest -Uri "http://localhost:8080/PowerShell/?test=linkheader&maxlinks=5"
+    #>
+    Param (
+        [Parameter()]
+        [Int] $Port = 8080,
+
+        [Parameter()]
+        [switch] $AsJob
+        )
+
+    Process {
+        $ErrorActionPreference = "Stop"
+
+        [scriptblock]$script = {
+            param ($Port)
+
+            # HttpListener.QueryString is not being populated, need to follow-up with CoreFx, workaround is to parse it ourselves
+            Function ParseQueryString([string]$url)
+            {
+                Write-Verbose "Parsing: $url"
+                $queryItems = @{}
+                foreach ($segment in $url.Split("&"))
+                {
+                    if ($segment -match "\??(?<name>\w*)(=(?<value>.*?))?$")
+                    {
+                        $name = $matches["name"]
+                        $value = $matches["value"]
+                        if ($value -ne $null)
+                        {
+                            $value = [System.Web.HttpUtility]::UrlDecode($value)
+                        }
+                        $queryItems.Add($name, $value)
+                        Write-Verbose "Found: $name = $value"
+                    }
+                }
+                return $queryItems
+            }
+
+            $listener = [System.Net.HttpListener]::New()
+            $urlPrefix = "/PowerShell"
+            $url = "http://localhost:$Port$urlPrefix/"
+            $listener.Prefixes.Add($url)
+            $listener.AuthenticationSchemes = [System.Net.AuthenticationSchemes]::Anonymous 
+            try {
+                Write-Warning "Note that thread is blocked waiting for a request.  After using Ctrl-C to stop listening, you need to send a valid HTTP request to stop the listener cleanly."
+                Write-Warning "Sending 'exit' command will cause listener to stop"
```

Comment:
Instead of this "Sending 'exit' command" instruction I would rather see something that could be copy/pasted like:
Stop-HTTPListener / Invoke-WebRequest -Uri "http://localhost:$port/PowerShell?test=exit"